src/annotator/guest.ts

```diff
@@ -706,7 +707,16 @@ export class Guest extends TinyEmitter implements Annotator, Destroyable {
    * Show or hide the adder toolbar when the selection changes.
    */
   _onSelection(range: Range) {
-    if (!this._integration.canAnnotate(range)) {
+    let textRange: Range;
+    try {
+      textRange = TextRange.fromRange(range).toRange();
+      textRange = trimRange(textRange);
```

Comment:
I suggest we add a short comment here to indicate reasons why we are trimming the text range.